src/api/route-services/access/index.js

```diff
@@ -99,6 +99,74 @@ class AccessService {
 
     return projectIds;
   }
+
+  // eslint-disable-next-line class-methods-use-this
+  async getExperimentEntries(table, experimentId) {
+    const docClient = new AWS.DynamoDB.DocumentClient({
+      region: config.awsRegion,
+    });
+    const getParams = {
+      TableName: table,
+      KeyConditionExpression: 'experimentId = :hkey',
+      ExpressionAttributeValues: {
+        ':hkey': experimentId,
+      },
+    };
+
+    const response = await docClient.query(getParams).promise();
+    return response.Items;
+  }
+
+  async deleteExperimentEntries(table, experimentId, items) {
+    const docClient = new AWS.DynamoDB.DocumentClient({
+      region: config.awsRegion,
+    });
+
+    // the user access table uses userId as key
+    // the invite access table uses userEmail as key
+    const userMapper = (item) => ({ userId: item.userId });
+    const inviteMapper = (item) => ({ userEmail: item.userEmail });
+    // have a different mapper for each of them so we can reuse the rest of the delete logic
+    const mapper = table === this.inviteAccessTableName ? inviteMapper : userMapper;
+
+    // create the batch delete request for each user that had permissions for the experiment
+    const deleteRequests = [];
+    items.forEach((item) => {
+      deleteRequests.push({
+        DeleteRequest: {
+          Key: {
+            experimentId,
+            ...mapper(item),
+          },
+        },
+      });
+    });
+
+    const deleteParams = {
+      RequestItems: {
+        [table]: deleteRequests,
+      },
+    };
+
+    await docClient.batchWrite(deleteParams).promise();
+  }
+
+  async deleteByExperimentId(table, experimentId) {
+    const items = await this.getExperimentEntries(table, experimentId);
+    if (items.length === 0) {
+      console.log(`no permissions rows found in ${table} for ${experimentId}`);
```

Comment:
why don't you use the logger for this message?

Reply:
I forgot to delete this